1-Draft/RFCnnnn-Alias-Policy.md

```diff
@@ -0,0 +1,108 @@
+
+    It is even more unpredictable and dangerous to create scripts in the assumption that there are some predefined (standard) aliases because they can be changed by the user or script at any time.
+
+    Modern versions of ScriptAnalyzer already recommend replacing aliases in scripts with their full names.
+
+    >The definition of aliases in modules or scripts and their automatic import becomes an undesirable and dangerous practice.
+
+1. UX in interactive sessions.
+
+    Having many cmdlets and parameters with long names can make it difficult to enter them in interactive sessions.
+    Users who typically use standard settings are typically satisfied with the ```IntellySense``` capabilities.
+    On the other hand, some users actively using interactive sessions may still be annoyed by the need to type names.
+    These users should be able to upload their familiar and useful aliases sets to their interactive sessions.
+    Given the limited functionality of aliases, these users will also configure their environment by executing commands by importing their own functions and various modules using PowerShell profiles.
+    It should be noted that we are planning to add a domain-specific language (DSL). This will provide users with much greater customization capabilities than aliases.
+
+# PowerShell alias policy
+
+1. PowerShell implements standard aliases in all sessions including interactive:
+
+    * % -> ForEach-Object
+    * foreach -> ForEach-Object
+    * ? -> Where-Object
+    * where -> Where-Object
+
+    These aliases are actively used in interactive sessions and scripts.
+    ScriptAnalyzer will continue to recommend not to use them in scripts and modules.
+
+1. By default, PowerShell does not load aliases from modules and InitialSessionState.
+
+    To ensure module backward compatibility:
+    * ```Import-Module -Aliases``` - explicitly loads aliases from the module.
+    * Global configuration option ```GlobalAliasImport``` in ```powershell.config``` turns on loading aliases from modules (and InitialSessionState). It is turned off by default.
+
+1. Backward compatibility of standard aliases.
+
+    To ensure backward compatibility, three sets of aliases are created:
+    * ```Default``` - current alias set (from ```InitialSessionState```) in PowerShell Core. This is the minimum set of aliases that have no conflicts on UNIX systems.
+    * ```Extended``` - the ```Default``` set of aliases and additional popular aliases for Windows users (which can have conflicts on Unix).
+    * ```WindowsPowerShell``` - the full set of standard aliases from Windows PowerShell. This set provides full backward compatibility for Windows users.
+
+    No alias set are loaded by default.
+
+    Global configuration option ```GlobalAliasSet``` in ```powershell.config``` turns on loading the desired set. It is turned off by default.
+
+1. PowerShell prevents the use of aliases in scripts and modules.
```

Comment:
This would probably be the single biggest breaking change ever proposed.

Reply:
This surprised me as well. I assume this means the inverse (to preserve back-compat all aliases are loaded by default). Separate to this is perhaps an ignorant (but important to me) question: Does my powershell.config live next to my profile.ps1? I would like a way from my user session to ensure no PowerShell "builtin" aliases are included in my interactive session without scrubbing them by hand in my profile, but I don't want to change machine state and I don't want to have to do this manually on every machine I spin up (I auto-sync profile.ps1 today, and I want a mechanism that lets me auto-sync these settings alongside)

Reply:
powershell.config(.json) lives in the install folder of PowerShell Core 

Currently it has to be re-edited every couple months when you deploy the latest PowerShell. That's one of the reasons I don't want this sort of thing in there 😉

Reply:
Agree this is a massive breaking change.

I am also concerned with the suggestion of powershell.config(.json) as it is a per install not a per-user setting. Some users might want different alias options if this change was to go forward, it would also require more deployment work for each new PS Core release.

Reply:
Of course, @kjacobsen -- the overall intention is to make it so that the only aliases present are the ones the user wants. They can choose to use one of the default sets, or create their own sets. If the choice is managed in your profile --or a unified config location-- there should be no repeating deployment work, you'd just have a one-time need to create or modify a profile, right?